### ISSUE TITLE
class org.asciidoctor.jruby.ast.impl.CellImpl cannot be cast to class org.asciidoctor.ast.StructuralNode

### ISSUE THREAD

**Issue by rahmanusta** [asciidoctor/asciidoctorj#1096]
Hello,

I'm using the findBy API, and I'm getting the following error.

```
java.lang.ClassCastException: class org.asciidoctor.jruby.ast.impl.CellImpl cannot be cast to class org.asciidoctor.ast.StructuralNode (org.asciidoctor.jruby.ast.impl.CellImpl and org.asciidoctor.ast.StructuralNode are in unnamed module of loader 'app')
	at com.kodedu.service.extension.CacheAppendRemoverProcessor.process(CacheAppendRemoverProcessor.java:23)
	at org.asciidoctor.jruby.extension.processorproxies.TreeprocessorProxy.process(TreeprocessorProxy.java:95)
	at org.asciidoctor.jruby.extension.processorproxies.TreeprocessorProxy$INVOKER$i$1$0$process.call(TreeprocessorProxy$INVOKER$i$1$0$process.gen)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodOneOrN.call(JavaMethod.java:967)
	at org.jruby.RubyMethod.call(RubyMethod.java:119)
```

The processor that used below:

```
public class CacheAppendRemoverProcessor extends Treeprocessor {
    private final Map<Object, Object> SELECTOR = Map.of("traverse_documents", "true");

    @Override
    public Document process(Document document) {
        List<StructuralNode> nodes = document.findBy(SELECTOR);
        for (StructuralNode node : nodes) {
            if (node instanceof BlockImpl block) {
                if ("image".equals(block.getContext())) {
                    String target = (String) block.getAttribute("target");
                    if (Objects.nonNull(target)) {
                        String[] split = target.split("\\?cache");
                        if (split.length == 2) {
                            block.setAttribute("target", split[0], true);
                        }
                    }
                }
            }
        }
        return document;
    }

}
```

Is it a bug or do I miss something?

Thank you

**Comment by abelsromero** [comment]
It seems there's a table in the doc that needs to be treated. Can you share a document to reproduce please?

**Comment by rahmanusta** [comment]
I'm able to reproduce it when a doc have table. Example below.

Test.adoc
```
== Test


[width="100%",options="header,footer"]
|====================
| cell | cell | cell 
| cell | cell | cell 
| cell | cell | cell 
|====================
```

**Comment by abelsromero** [comment]
I see the issue is `findBy` supposed to return `List<StructuralNode>`, but it also returns the table cells whose class does not extend from that. 
For now to be safe, you could cast the return to `ContentNode` instead and treat each node accordingly.

**Comment by robertpanzer** [comment]
I wonder why we have made Cell a ContentNode instead of a StructuralNode.
In Asciidoctor Ruby a Cell is an AbstractBlock just like a "normal" Block or a Table.
I'll play with making it a subclass of a StructuralNode.

**Comment by robertpanzer** [comment]
@rahmanusta I just published AsciidoctorJ 2.5.5 that should fix this bug.
